### PR TITLE
[api] include `mLinkMargin` in `otNeighborInfo`

### PR DESCRIPTION
**Pull request by Mikeadaneg**
Included mLinkMargin in the otNeighborInfo struct to retrieve Link margin via otThreadGetNextNeighborInfo

### CODE REVIEW

File: include/openthread/thread.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ typedef struct
     uint8_t      mLinkQualityIn;        ///< Link Quality In
     int8_t       mAverageRssi;          ///< Average RSSI
     int8_t       mLastRssi;             ///< Last observed RSSI
+    uint8_t      mLinkMargin;           ///< Link Margin
```
Comment (abtink) [inline include/openthread/thread.h:100]: Thanks @Mikeadaneg. Looks good.
You also need to increment API version number in `include/openthread/instnace.h`.

Comment (Mikeadaneg) [reply include/openthread/thread.h:100]: @abtink Just updated the API version number to 281

**Review by abtink — APPROVED** [review]
Thanks.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**